dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/BaseClasses.mo

```diff
@@ -54,13 +53,196 @@ package BaseClasses
+
+    annotation(preferredView = "text");
+  end BasePQProp;
+
+  partial model BasePTanPhiDangling "Base dynamic model for P/tan(Phi) control at terminal 1"
+
+    input Real tanPhi1Ref(start = TanPhi1Ref0) "tan(Phi) regulation set point at terminal 1";
+
+    parameter Real TanPhi1Ref0 "Start value of tan(Phi) regulation set point at terminal 1";
+
+  protected
+    Types.ReactivePowerPu QInj1RawPu "Raw reactive power at terminal 1 in pu (base SnRef) (generator convention)";
+
+    annotation(preferredView = "text");
+  end BasePTanPhiDangling;
+
+  partial model BasePTanPhi "Base dynamic model for P/tan(Phi) control"
+    extends BaseClasses.BasePTanPhiDangling;
+
+    input Real tanPhi2Ref(start = TanPhi2Ref0) "tan(Phi) regulation set point at terminal 2";
+
+    parameter Real TanPhi2Ref0 "Start value of tan(Phi) regulation set point at terminal 2";
+
+  protected
+    Types.ReactivePowerPu QInj2RawPu "Raw reactive power at terminal 2 in pu (base SnRef) (generator convention)";
+
+    annotation(preferredView = "text");
+  end BasePTanPhi;
+
+  partial model BasePVDangling "Base dynamic model for PV control at terminal 1"
+
+    parameter Boolean UseLambda1 = not(Lambda1Pu == 0) "If true, the voltage regulation follows the law U1RefPu = U1Pu + Lambda1Pu * QInj1Pu at terminal 1";
+    parameter Types.PerUnit Lambda1Pu "Parameter Lambda of the voltage regulation law U1RefPu = U1Pu + Lambda1Pu * QInj1Pu, in pu (base UNom, SnRef) at terminal 1";
+    parameter Types.ReactivePower Q1Nom "Nominal reactive power in Mvar at terminal 1";
+
+    input Types.VoltageModulePu U1RefPu(start = U1Ref0Pu) "Voltage regulation set point in pu (base UNom) at terminal 1";
+
+    parameter Types.VoltageModulePu U1Ref0Pu "Start value of the voltage regulation set point in pu (base UNom) at terminal 1";
+
+    Types.ReactivePowerPu QInj1PuQNom "Reactive power at terminal 1 in pu (base Q1Nom) (generator convention)";
+
+    annotation(preferredView = "text");
+  end BasePVDangling;
+
+  partial model BasePV "Base dynamic model for PV control"
+    extends BaseClasses.BasePVDangling;
+
+    parameter Boolean UseLambda2 = not(Lambda2Pu == 0) "If true, the voltage regulation follows the law U2RefPu = U2Pu + Lambda2Pu * QInj2Pu at terminal 2";
+    parameter Types.PerUnit Lambda2Pu "Parameter Lambda of the voltage regulation law U2RefPu = U2Pu + Lambda2Pu * QInj2Pu, in pu (base UNom, SnRef) at terminal 2";
+    parameter Types.ReactivePower Q2Nom "Nominal reactive power in Mvar at terminal 2";
+
+    input Types.VoltageModulePu U2RefPu(start = U2Ref0Pu) "Voltage regulation set point in pu (base UNom) at terminal 2";
+
+    parameter Types.VoltageModulePu U2Ref0Pu "Start value of the voltage regulation set point in pu (base UNom) at terminal 2";
+
+    Types.ReactivePowerPu QInj2PuQNom "Reactive power at terminal 2 in pu (base Q2Nom) (generator convention)";
```

Comment:
This is no initial variable, so this line has to be moved between U2RefPu and U2Ref0Pu.